client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -58,34 +58,35 @@ const HostingFeatures = () => {
 	);
 	const hasEnTranslation = useHasEnTranslation();
 
-	const transfer = useSelector( ( state ) => getAtomicTransfer( state, siteId ) );
-	const isTransferInProgress = [
-		transferStates.PENDING,
-		transferStates.ACTIVE,
-		transferStates.PROVISIONED,
-	].includes( transfer.status );
+	const { data: siteTransferData, refetch: refetchSiteTransferData } = useSiteTransferStatusQuery(
+		siteId || undefined
+	);
+	// We have some bug at the backend, when the transfer is complited, but isSiteAtomic is still false, so we need such extra condition
+	const isTransferInProgress =
+		siteTransferData?.isTransferring || siteTransferData?.status === transferStates.COMPLETED;
 
 	useEffect( () => {
 		if ( ! siteId ) {
 			return;
 		}
 
-		if ( transfer.status === transferStates.COMPLETED ) {
-			window.location.href = `/overview/${ siteSlug }`;
-			return;
-		}
-
-		dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		// A user can keep one tab open and start transfer in the second tab
-		// se we always run interval to check the status, to avoid case with rendering "Activate" button on old tabs of a browser
 		const interval = setInterval( () => {
-			dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
-		}, 10000 );
+			if ( siteTransferData?.status !== transferStates.COMPLETED ) {
+				refetchSiteTransferData();
+			} else {
+				clearInterval( interval );
+				dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+			}
+		}, 3000 );
 
-		return () => {
-			clearInterval( interval );
-		};
-	}, [ siteSlug, siteId, transfer.status, dispatch ] );
+		return () => clearInterval( interval );
+	}, [ siteId, siteTransferData?.status, refetchSiteTransferData, dispatch ] );
+
+	useEffect( () => {
+		if ( isSiteAtomic && ! isPlanExpired ) {
```

Comment:
```suggestion
		if ( isSiteAtomic && ! isTransferInProgress && ! isPlanExpired ) {
```
I want to be minimally invasive with my review, but did you consider doing this, @nightnei? Would it make a difference?

Reply:
I doesn't make any difference since every time when I tested - `isTransferInProgress ` was always true, even when `isSiteAtomic `. We get always `isTransferInProgress` before `isSiteAtomic`.

I will double check it in the follow up PR, when I will be doing polishments